File: JavaScript/Core1/InventoryManagement/SingleProductShop.js
class SingleProductShop extends oBJECT {
  constructor({ id, name, quantityInStock, reorderLevel, targetInventory, reorderInterval}) {
    super( id, name);
    this.quantityInStock = quantityInStock;
    this.reorderLevel = reorderLevel;  // for continuous review policy
    this.targetInventory = targetInventory;
    this.reorderInterval = reorderInterval;  // for periodic review policy
  }
}
SingleProductShop.labels = {"quantityInStock":"stock"};

File: Java/Core1/src/main/java/de/oes/core1/inventorymanagment/SingleProductShop.java
package de.oes.core1.inventorymanagment;


import de.oes.core1.foundations.oBJECT;
import de.oes.core1.sim.Simulator;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class SingleProductShop extends oBJECT{

	private Integer quantityInStock;
	private Integer reorderPoint; // for continuous review policy
	private Integer targetInventory;
	private Integer reorderInterval; // for periodic review policy
	
	public SingleProductShop(Integer id, String name, Simulator sim, int quantityInStock, int reorderPoint, int targetInventory,  int reorderInterval) {
		super(id, name, sim);
		this.quantityInStock = quantityInStock;
		this.reorderPoint = reorderPoint;
		this.targetInventory = targetInventory;
		this.reorderInterval = reorderInterval;
	}

	@Override
	public String toString() {
		return this.getClass().getSimpleName() + "{stock : " + this.getQuantityInStock() + "}";
	}
}
